### DISCUSSION TITLE
For pypi datasources, only 'project_urls' info is used for lookup, not 'project_url'

### DISCUSSION THREAD

**Discussion by nico-kn** [renovatebot/renovate#9907] [Suggest an Idea]
**What Renovate type, platform and version are you using?**

**Renovate type**

- [ ] Hosted app
- [x] Self hosted

**Renovate platform**

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [x] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Renovate version (if self hosting)**

I'm self hosting Renovate with version: 24.6

**Describe the bug**

Renovate will not recognize the dependency's project infos if the attribute name differs from "project_urls". That means that changelog and version infos will not be part of the PR description although they are not excluded by config, see screenshot.

I am using renovate to update `pip` dependencies, that are self-hosted on an internal registry. The internal registry is [Sonatype Nexus](https://help.sonatype.com/repomanager3/formats/pypi-repositories) that mirrors pypi. Nexus will display the `project_urls` info attribute as `project_url` which, I assume, is not recognized by renovate, see screenshot attached. My assumption how this could be fixed can be found in "Additional Context" below.

![image](https://user-images.githubusercontent.com/22818871/117465027-8a27ad80-af51-11eb-9580-72cc4dbf74a4.png)
![image](https://user-images.githubusercontent.com/22818871/117464830-5482c480-af51-11eb-988a-a1fccb1a03d5.png)

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

I ran the renovate bot with DEBUG enabled, however no usage of "getReleases" or "getDependencies" is logged at all. I will happily provide logs if you let me know what to look out for.

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly (due to Nexus as registry)

**Additional context**

I assume that adding an alias for possible values of `project_urls` would easily fix this issue to support other third party pypi mirrors as well. This could be done in https://github.com/renovatebot/renovate/blob/main/lib/datasource/pypi/index.ts#L76

**Comment by rarkins** [comment]
Based on your screenshot, is this field unstructured text?

**Comment by nico-kn** [comment]
> Based on your screenshot, is this field unstructured text?

@rarkins yes, exactly. Below are two examples, copied as html fragments from the Nexus UI.

This is the example for `requests`:
```
<html>
<body>
<!--StartFragment-->

project_url | Documentation, https://requests.readthedocs.io Source, https://github.com/psf/requests
-- | --


<!--EndFragment-->
</body>
</html>
```
And this for `aiohttp`: 
```
<html>
<body>
<!--StartFragment-->

project_url | Chat: Gitter, https://gitter.im/aio-libs/Lobby CI: Azure Pipelines, https://dev.azure.com/aio-libs/aiohttp/_build Coverage: codecov, https://codecov.io/github/aio-libs/aiohttp Docs: RTD, https://docs.aiohttp.org GitHub: issues, https://github.com/aio-libs/aiohttp/issues GitHub: repo, https://github.com/aio-libs/aiohttp
-- | --


<!--EndFragment-->
</body>
</html>
```

**Comment by rarkins** [comment]
Assuming the correct interpretation of the above is that it's a non standard and/or unstructured fiend, I think this will require someone like yourself with Nexus who can implement and test extensively locally before submitting a PR, otherwise it won't be implemented.

**Comment by rarkins** [comment]
Converting this to a discussion as it's currently misclassified as a bug
